File: backendworker/NHSOnline.Backend.GpSystems/Suppliers/Tpp/Models/Appointments/Session.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.CodeAnalysis;
using System.Xml.Serialization;

namespace NHSOnline.Backend.GpSystems.Suppliers.Tpp.Models.Appointments
{
    [SuppressMessage("Microsoft.Naming", "CA1724", Justification = "Deliberately matching the name specified by the GPSS")]
    [Serializable]
    public class Session
    {
        [XmlAttribute("sessionId")]
        public string SessionId { get; set; }

        [XmlAttribute("type")]
        public string Type { get; set; }

        [XmlAttribute("staffDetails")]
        public string StaffDetails { get; set; }

        [XmlAttribute("location")]
        public string Location { get; set; }

        [XmlElement("Slot", typeof(Slot))]
        public List<Slot> Slots { get; set; }
    }
}

File: bddtests/src/main/kotlin/mocking/tpp/models/Session.kt
package mocking.tpp.models

import javax.xml.bind.annotation.XmlAccessType
import javax.xml.bind.annotation.XmlAccessorType
import javax.xml.bind.annotation.XmlAttribute
import javax.xml.bind.annotation.XmlElement
import javax.xml.bind.annotation.XmlRootElement

@XmlRootElement(name = "Session")
@XmlAccessorType(XmlAccessType.FIELD)
data class Session (
    @XmlAttribute var sessionId: String = "default sessionId",
    @XmlAttribute var type: String = "default type",
    @XmlAttribute var staffDetails: String = "default staffName",
    @XmlAttribute var location: String = "default location",
    @XmlElement var Slot: MutableCollection<Slot> = mutableListOf()
)
